dan_layer/consensus/src/hotstuff/pacemaker.rs

```diff
@@ -20,24 +21,31 @@ use crate::hotstuff::{
 
 const LOG_TARGET: &str = "tari::dan::consensus::hotstuff::pacemaker";
 const MAX_DELTA: Duration = Duration::from_secs(300);
-const BLOCK_TIME: Duration = Duration::from_secs(10);
+const DEFAULT_BLOCK_TIME: Duration = Duration::from_secs(10);
 
 pub struct PaceMaker {
     pace_maker_handle: PaceMakerHandle,
     handle_receiver: mpsc::Receiver<PacemakerRequest>,
     current_view: CurrentView,
     current_high_qc_height: NodeHeight,
+    block_time: Duration,
 }
 
 impl PaceMaker {
-    pub fn new() -> Self {
+    pub fn new(max_block_time_threshold: u64) -> Self {
         let (sender, receiver) = mpsc::channel(100);
 
         let on_beat = OnBeat::new();
         let on_force_beat = OnForceBeat::new();
         let on_leader_timeout = OnLeaderTimeout::new();
         let current_height = CurrentView::new();
 
+        let block_time = if max_block_time_threshold == 0 {
+            DEFAULT_BLOCK_TIME
+        } else {
+            Duration::from_secs(max_block_time_threshold)
+        };
+
```

Comment:
If set to Duration::from_secs(0), it can stay like that (yeah it could cause problems but then again it may be unexpected that it changes to 10 by itself, maybe we want to set 0 i.e. 2 sec block time). The default in the consensus constants will already be set to 10 by default if it is omitted.